src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -95,6 +93,7 @@ protected virtual Task HandleReceiveExceptionAsync(Exception receiveException)
         /// </summary>
         /// <param name="messageHandler">The message handler to be used to process the message.</param>
         /// <param name="messageContext">The message context of the message that will be handled.</param>
+        [Obsolete("Consider using the " + nameof(IServiceCollectionExtensions.AddMessageRouting) + " instead to register an " + nameof(IMessageRouter) + " implementation that you can use directly in your message pump")]
```

Comment:
The `MessagePump` is an abstraction (`abstract`) so can't be added to the application by itself. It has to be added via a dependent class (like `AzureServiceBusMessagePump`). While registering the Azure Service Bus message pump, we can behind the scenes register our router, but since we don't have this opportunity with the `abstract` `MessagePump`, we can't do that. These `[Obsolete]` methods can still be used though, if you get (lazy) our `IMessageRouter` from the services.